### PR TITLE
mvn optimization for 1D and 2D input

### PR DESCRIPTION
**Pull request by chenhu-wang**
### Details:
 - *1D and 2D input optimization*

### Tickets:
 - *52820*

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@a-sidorova  Please review the PR

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_mvn_node.cpp [hunk 1]
```diff
@@ -643,6 +643,38 @@ void MKLDNNMVNNode::getSupportedDescriptors() {
     } else if (details::CaselessEq<std::string>()(epsMode, "outside_sqrt")) {
         epsMode_ = outsideSqrt;
     }
+
+    SizeVector inputShape = getParentEdgeAt(0)->getDims().ToSizeVector();
```
Comment (a-sidorova) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_mvn_node.cpp:647]: Minor suggestion - add 626th line:
`const SizeVector inputShape = getParentEdgeAt(0)->getDims().ToSizeVector();`
And change 627th line to:
`const auto& numOfDims = inputShape.size();`

**Review by a-sidorova — APPROVED** [review]

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_mvn_node.cpp (hunk 1 above)
Comment (chenhu-wang) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_mvn_node.cpp]: done, thank you.

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_mvn_node.cpp [hunk 2]
```diff
@@ -506,7 +506,7 @@ struct jit_uni_mvn_kernel_f32 : public jit_uni_mvn_kernel, public jit_generator
     Xbyak::Reg64 reg_d_bias = rdx;
 
     Xbyak::Reg64 reg_load_table = r15;
-    Xbyak::Reg64 reg_load_store_mask = rcx;
+    Xbyak::Reg64 reg_load_store_mask = rbp;
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_mvn_node.cpp:509]: Just wondering how the impl on master is working on Windows? Do we cover this path by tests?

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_mvn_node.cpp [hunk 3]
```diff
@@ -690,6 +691,37 @@ MKLDNNMVNNode::MKLDNNMVNNode(const std::shared_ptr<ngraph::Node>& op, const mkld
...
+                shape5D = std::make_tuple(1, 1, 1, 1, shape[0]);
+                acrossChannels_ = false;
+                break;
+            } else {
+                shape5D = std::make_tuple(1, shape[0], 1, 1, 1);
+                break;
+            }
+        case 2 :  // NC
+            if (acrossChannels_) {
+                shape5D = std::make_tuple(1, shape[0], 1, shape[1], 1);
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_mvn_node.cpp:713]: Why this case is not processed in the same manner as case 1 with acrossChannels_ ("if acrossChannels_ is true, adjust shape to fully vectorize under unified 5d procedure")? Or after normalization both 3rd and 4th dimensions will be used for vectorization?

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_mvn_node.cpp (hunk 2 above)
Comment (chenhu-wang) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_mvn_node.cpp:509]: Thank you for this question and I re-checked the code.
reg_load_store_mask is only used on avx512 target for [mask load/store](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/emitters/jit_load_store_emitters.cpp#L158) for tails.
Other target use [progressive load/store](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/emitters/jit_load_store_emitters.cpp#L200) for tails.
So we just care about avx512. On this target rcx used and only used as abi_param1 at the very beginning of execution loop to load parameters. Then we can reuse rcx in the execution loop within emit_code() to load data.
Tests already cover this case such as input tensor of [dimension [2, 19, 5, 10]](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/functional/plugin/cpu/single_layer_tests/mvn.cpp#L88), 5*10 is not devisable by 16, which is processed in one kernel in planar layout. The mask loaded for tails on avx512 happens here.
So `reg_load_store_mask = rcx` does not lead to bug, changing just for remove possible worries and readability and meanwhile rbp is available in this node.

Comment (dmitry-gorokhov) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_mvn_node.cpp:509]: I see. Thanks for explanation.

**Pull request merged**